RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -252,19 +412,26 @@ EvtPlaneProducer::EvtPlaneProducer(const edm::ParameterSet &iConfig)
 
   vertexToken = consumes<std::vector<reco::Vertex>>(vertexTag_);
 
-  caloToken = consumes<CaloTowerCollection>(caloTag_);
-
-  castorToken = consumes<std::vector<reco::CastorTower>>(castorTag_);
-
-  trackToken = consumes<reco::TrackCollection>(trackTag_);
+  strack = trackTag_.label();
+  if (strack.find("packedPFCandidates") != std::string::npos) {
```

Comment:
please replace strack use with a boolean or a named enum if multiple states are possible; this will avoid need for string operations in every event on something that's defined at construction/configuration step